tests/callbacks/test_rich_progress_bar.py

```diff
@@ -141,3 +141,24 @@ def on_train_start(self) -> None:
 
         trainer.fit(model)
     mock_progress_stop.assert_called_once()
+
+
+@RunIf(rich=True)
+def test_rich_progress_bar_configure_columns(tmpdir):
+    from rich.progress import TextColumn
+
+    custom_column = TextColumn("[progress.description]Testing Rich!")
+
+    class CustomRichProgressBar(RichProgressBar):
+        def configure_columns(self, trainer, pl_module):
+            return [custom_column]
+
+    progress_bar = CustomRichProgressBar()
+
+    model = BoringModel()
+    trainer = Trainer(default_root_dir=tmpdir, fast_dev_run=True, callbacks=progress_bar)
+
+    trainer.fit(model)
+
+    assert progress_bar.progress.columns[0] == custom_column
+    assert len(progress_bar.progress.columns) == 1
```

Comment:
Not sure how much work it is, but we could convert the need to run this end to end into a unit test potentially? might be able to just instantiate CustomRichProgressBar and call `configure_columns` passing a trainer/model object and assert the progress bar contains the column?

Albeit this doesn't test the full end to end logic ensuring that we call this in the right place when trainer.fit is called

Reply:
Good point, @SeanNaren Updated!